README.md

```diff
@@ -64,17 +64,30 @@ class MyDataSource(panoply.DataSource):
 
 The `DataSource` base class exposes the following methods:
 
-###### __init__(self, source, options)
+###### __init__(self, source, options, events)
 
-Constructor. Receives a dictionary with the data `source` parameters (see below) and a dictionary with any additional `options`. Generally, it's safe to disregard the options, however it may be used for performance optimizations, as it contains hints about incremental keys, excluded fields, etc.
+Constructor. Receives a dictionary with the data `source` parameters (see below), a dictionary with any additional `options` and a dictionary with `events` subscriptions. Generally, it's safe to disregard the options, however it may be used for performance optimizations, as it contains hints about incremental keys, excluded fields, etc, and it may contain additional parameters that can't be transferred with the source
```

Comment:
In Google Drive (like in some other streams), the user is asked to select files before the collect process.
This is done, by using the google drive API that relies on a valid token and that token could be invalidated, meaning that it might change before the `read` process (e.g. `files = lambda s, o: Stream(s, o).get_files()`)

Before, `source-change` events could be fired by the `read` method in order to allow saving changes in the source but since this event was only attached in `start` under `runner.py` after the constructor finishes its operation, there was no way to explicitly save changes in the source from within the constructor and before the `collect` process runs.

The new way of allowing this (and a better approach I think) in the above given example is to use `lambda *args, **kwargs: Stream(*args, **kwargs).get_files()` - this will ensure that the events dictionary will be passed to the stream and that future such changes will not require additional changes in the data sources.